nixos/modules/services/security/oauth2_proxy.nix

```diff
@@ -494,10 +506,43 @@ in
       '';
     };
 
+    setXauthrequest = mkOption {
+      type = types.nullOr types.bool;
+      default = null;
+      description = ''
+        Set X-Auth-Request-User and X-Auth-Request-Email response headers (useful in Nginx auth_request mode).
+      '';
+    };
```

Comment:
Document the fact that `null` means not passing the option to the binary in order to let it use the upstream default.